pkg/operator/ceph/cluster/osd/key_rotation.go

```diff
@@ -0,0 +1,289 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package osd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	kms "github.com/rook/rook/pkg/daemon/ceph/osd/kms"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	keyRotationCronJobAppName    = "rook-ceph-osd-key-rotation"
+	keyRotationCronJobAppNameFmt = "rook-ceph-osd-key-rotation-%d"
+)
+
+// keyRotationCronJobName returns the name of the key rotation cron job for the given OSD ID.
+func keyRotationCronJobName(osdID int) string {
+	return fmt.Sprintf(keyRotationCronJobAppNameFmt, osdID)
+}
+
+// applyKeyRotationPlacement applies the placement settings for the key rotation job
+// so that it is scheduled on the same node as the pod containing the given labels.
```

Comment:
nit: This comment would help be clear about the intent of running with the OSD.
```suggestion
// so that it is scheduled on the same node as the OSD for which the key rotation is scheduled.
```